icu4j/main/classes/core/src/com/ibm/icu/impl/number/MixedUnitLongNameHandler.java

```diff
@@ -0,0 +1,185 @@
+     */
+    @Override
+    public MicroProps processQuantity(DecimalQuantity quantity) {
+        assert (fMixedUnitData.size() > 1);
+        MicroProps micros;
+        // if (parent != null)
+        micros = parent.processQuantity(quantity);
+        micros.modOuter = getMixedUnitModifier(quantity, micros);
+        return micros;
+    }
+
+    // Required for ModifierStore. And ModifierStore is required by
+    // SimpleModifier constructor's last parameter. We assert his will never get
+    // called though.
+    public Modifier getModifier(Modifier.Signum signum, StandardPlural plural) {
+        // TODO(units): investigate this method when investigating where
+        // LongNameHandler::getModifier() gets used. To be sure it remains
+        // unreachable:
+
+        return null;
+    }
+
+    // For a mixed unit, returns a Modifier that takes only one parameter: the
+    // smallest and final unit of the set. The bigger units' values and labels
+    // get baked into this Modifier, together with the unit label of the final
+    // unit.
+    private Modifier getMixedUnitModifier(DecimalQuantity quantity, MicroProps micros) {
+        // TODO(icu-units#21): mixed units without usage() is not yet supported.
+        // That should be the only reason why this happens, so delete this whole if
+        // once fixed:
+        if (micros.mixedMeasures.size() == 0) {
+            throw new UnsupportedOperationException();
+        }
+
+
+        // Algorithm:
+        //
+        // For the mixed-units measurement of: "3 yard, 1 foot, 2.6 inch", we should
+        // find "3 yard" and "1 foot" in micros.mixedMeasures.
+        //
+        // Obtain long-names with plural forms corresponding to measure values:
+        //   * {0} yards, {0} foot, {0} inches
+        //
+        // Format the integer values appropriately and modify with the format
+        // strings:
+        //   - 3 yards, 1 foot
+        //
+        // Use ListFormatter to combine, with one placeholder:
+        //   - 3 yards, 1 foot and {0} inches
+        //
+        // Return a SimpleModifier for this pattern, letting the rest of the
+        // pipeline take care of the remaining inches.
+
+        List<String> outputMeasuresList = new ArrayList<>();
+
+        for (int i = 0; i < micros.mixedMeasures.size(); i++) {
+            DecimalQuantity fdec = new DecimalQuantity_DualStorageBCD(micros.mixedMeasures.get(i).getNumber());
+            StandardPlural pluralForm = fdec.getStandardPlural(rules);
+
+            String simpleFormat = LongNameHandler.getWithPlural(this.fMixedUnitData.get(i), pluralForm);
+            SimpleFormatter compiledFormatter = SimpleFormatter.compileMinMaxArguments(simpleFormat, 0, 1);
+
+            FormattedStringBuilder appendable = new FormattedStringBuilder();
+            this.fIntegerFormatter.format(fdec).appendTo(appendable);
+            String num = compiledFormatter.format(outputMeasuresList.get(i));
+        }
```

Comment:
@hugovdm , @sffc ... is this part of the code correct, about using the append-able?